### PR TITLE
Add /rec to test OpenRefine Requests

### PR DESCRIPTION
**Pull request by GrantXie**
Handle get/post queries from OpenRefine and handle request callbacks for general information

### CODE REVIEW

File: wikifier_service.py [hunk 1]
```diff
@@ -6,13 +6,124 @@
...
 wikifier = Wikifier()
 config = json.load(open('wikifier/config.json'))
 
 
+
+
+@app.route('/rec', methods=['POST','GET'])
+def rec():
+   
+    #print(request.headers)
```
Comment (saggu) [inline wikifier_service.py:23]: delete commented out code

File: wikifier_service.py [hunk 2]
```diff
@@ -6,13 +6,124 @@
...
+    #print(request.url)
+    #print(request.__dict__)
+    #print(request.referrer)
+    #if flask.request.method == 'POST':
+    #    print(flask.request)
+
+
+    #deal with callback requests for general info
+
+    query = request.form.get('queries')
```
Comment (saggu) [inline wikifier_service.py:36]: query should either be in form or args, should not check both

File: wikifier_service.py [hunk 3]
```diff
@@ -6,13 +6,124 @@
...
+
+    if query == None:
+         query = request.args.get('queries')
+    
+    if query == None:
+        callback = request.args.get('callback', False)
+        if callback:
+            content = str(callback) + '(' + str({
+           "name" : "Table-Linker Reconciliation for OpenRefine (en)",
```
Comment (saggu) [inline wikifier_service.py:45]: why do you need wikidata urls?

**Review by saggu — CHANGES_REQUESTED** [review]
The code is not conforming to PEP 8 standards, please check. Have you tried to call this service from openrefine web app?

File: wikifier_service.py (hunk 3 above)
Comment (GrantXie) [reply wikifier_service.py:45]: the wikidata urls are for openrefine to put link on the candidates

File: wikifier_service.py [hunk 4]
```diff
@@ -6,13 +6,101 @@
...
+                df = df.drop('type_strict', 1)
+
+
+        _uuid_hex = uuid4().hex
+
+        _path = 'user_files/{}_{}'.format(columns, _uuid_hex)
+        pathlib.Path(_path).mkdir(parents=True, exist_ok=True)
+
+        wikifier.wikify(df, columns, output_path=_path, debug=True, k=k,
+                                      colorized_output='test1.csv')
```
Comment (saggu) [inline wikifier_service.py:79]: colorized_output is a boolean parameter

File: wikifier_service.py [hunk 5]
```diff
@@ -6,13 +6,101 @@
...
+        for i in range(0, len(df)):
+            output[label[df['row'][i]]]['result'].append({
+                "id" : df['kg_id'][i],
+                "name" : df['kg_labels'][i],
+                "type" : [{"id":"/qnode","name":"Qnode"}],
+                "score" : df['siamese_prediction'][i],
+                "match" : (float(df['siamese_prediction'][i])> 0.95 and int(df['rank'][i]) == 1)
+              })
+        
+        callback = request.args.get('callback', False)
```
Comment (saggu) [inline wikifier_service.py:97]: you have the code twice, no need to read `callback` again

**Pull request closed without merging**